services/app-web/src/components/FileUpload/FileItemList/FileItemsList.tsx

```diff
@@ -23,26 +26,51 @@ export const FileItemsList = ({
         })
     }
 
-    return (
-        <ul
-            data-testid="file-input-preview-list"
-            className={styles.fileItemsList}
-            id="file-items-list"
-            tabIndex={-1}
-        >
-            {fileItems.map((item) => (
-                <li
-                    key={item.id}
-                    id={item.id}
-                    className={liClasses(item.status)}
-                >
-                    <FileItem
-                        deleteItem={deleteItem}
-                        retryItem={retryItem}
-                        item={item}
-                    />
-                </li>
-            ))}
-        </ul>
-    )
+    if (renderMode === 'table') {
```

Comment:
I think that it might be cleaner here to use some composition instead of just a switch. I’m imagining new very small view-only components:

<FileItemsTableView>
<FileItemsListView>
<FileItemsTableRowView>
<FileItemsListRowView>

or something. And then you can pass the component in to <FileItemsList> ? 

If you want to pair on this and talk it through we could try and figure something out. I think we definitely still want to keep all the logic that is in FileItem.tsx solo there, so I don’t think it’s good to duplicate everything, but I think that we should be able to separate out these two views into their own things.

Reply:
I do wonder what it’ll look like once we have the checkboxes in the table view. then we’ll have controls in one that aren’t in the other.

Reply:
Yeah, I didn't love this, and in one of the many commits attached to this PR, there's a time when I had this broken out into separate components (fileItem and fileRow), but I ultimately undid that and opted for keeping the fileItem logic in one place.  I was torn about it, and am happy to look again to see if the components can be separated without maintaining the upload logic in two places.

Reply:
My take- the checkboxes will likely go inside the `FileItem` component since that is where the table rows and `<td>` are.  We need to contain form elements in a table inside a cell - that's an accessibility requirement.

I feel like that component (`FileItem`)  is a bit harder to read and has more essential logic than this one. `FileItemsList` is more of a  presentation component.